relayer/src/light_client/tendermint.rs

```diff
@@ -218,12 +224,49 @@ impl LightClient {
 
         let mut store = MemoryStore::new();
         store.insert(trusted_block, Status::Trusted);
+        tracing::warn!(trusted_height = %trusted, "finished fetching light block & inserted in block store");
 
         Ok(LightClientState::new(store))
     }
 
     fn fetch_light_block(&self, height: AtHeight) -> Result<LightBlock, Error> {
+        use core::time::Duration;
         use tendermint_light_client::components::io::Io;
+        use tracing::warn;
+
+        let _juno_client_latest_trusted_height = TMHeight::from(4136530_u32);
+        match height {
+            AtHeight::At(_juno_client_latest_trusted_height) => {
+                if self.chain_id == ChainId::new("juno".to_owned(), 1) {
+                    // Create an alternative io for the archive node, to bypass the default full node.
+                    warn!("matched on juno-1 and expected halt height");
+                    let archive_rpc_client =
+                        rpc::HttpClient::new("https://rpc-v3-archive.junonetwork.io:443")
+                            .expect("could not initialize the rpc client to bypass juno full node");
+                    // let peer_id_string = String::from("4b0f1f25d5bff62d6cd674ddbe56df14d58979f3").as_bytes();
+                    let dummy_peer_id_u8: [u8; 20] = [0; 20];
```

Comment:
It seems this hardcoded peer id is never problematic, likely it is not checked anywhere.

We might consider keeping this in mind in a light client refactor.